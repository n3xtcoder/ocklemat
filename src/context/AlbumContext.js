import React, { createContext, useState } from "react";

export const AlbumContext = createContext();

export const AlbumContextProvider = ({ children }) => {
  const [albums, setAlbums] = useState();

  const getData = async () => {
    try {
      const response = await fetch(
        "https://jsonplaceholder.typicode.com/albums/1/photos"
      );
      const obj = await response.json();
      // console.log(obj);
      setAlbums(obj);
    } catch (error) {
      console.log(error);
    }
  };

  return (
    <AlbumContext.Provider
      value={{
        albums,
        getData,
      }}
    >
      {children}
    </AlbumContext.Provider>
  );
};
