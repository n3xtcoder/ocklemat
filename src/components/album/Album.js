import React, { useContext, useEffect } from "react";
import { AlbumContext } from "../../context/AlbumContext";
import AlbumImage from "../albumImage/AlbumImage";
import "./album.css";
import CloseIcon from "@material-ui/icons/Close";
import BtnModal from "./BtnModal";
import { Link, Route, useHistory, Switch } from "react-router-dom";

function Album() {
  const { albums, getData } = useContext(AlbumContext);

  useEffect(() => {
    getData();
  }, []);

  ///////////
  //* Slider Controls
  ///////////

  const history = useHistory();

  const nextSlide = (index) => {
    if (index === albums.length - 1) {
      setHistory(albums[0].id);
    } else {
      setHistory(albums[index + 1].id);
    }
  };

  const prevSlide = (index) => {
    if (index === 0) {
      setHistory(albums[albums.length - 1].id);
    } else {
      setHistory(albums[index - 1].id);
    }
  };

  const setHistory = (id) => {
    history?.push({
      pathname: `/slider/${id}`,
      state: { modal: true },
    });
  };

  return (
    <div className="Album_Wrapper">
      {albums && (
        <Switch>
          {" "}
          <Route
            path="/slider/:id"
            children={({ match, history }) => {
              const id = +match?.params.id;
              const imageIndex =
                id && albums.findIndex((image) => image.id === id);
              if (imageIndex < 0) {
                return null;
              }

              const close = (e) => {
                e.stopPropagation();
                history.push("/");
              };

              return (
                <div className="modal open">
                  <div>
                    <img
                      className="modal-image"
                      src={albums[imageIndex].url}
                      alt={albums[imageIndex].title}
                    />

                    <CloseIcon className="modal-close-btn" onClick={close}>
                      close modal
                    </CloseIcon>

                    <BtnModal
                      moveSlide={() => nextSlide(imageIndex)}
                      direction={"next"}
                    />
                    <BtnModal
                      moveSlide={() => prevSlide(imageIndex)}
                      direction={"prev"}
                    />
                  </div>
                </div>
              );
            }}
          />
        </Switch>
      )}

      <div className="Album_GridContainer">
        {albums &&
          albums.map((item, index) => {
            return (
              <Link to={{ pathname: `/slider/${item.id}` }}>
                {" "}
                <AlbumImage
                  className="Album_gridImage"
                  key={index}
                  image={item}
                />
              </Link>
            );
          })}
      </div>
    </div>
  );
}

export default Album;
