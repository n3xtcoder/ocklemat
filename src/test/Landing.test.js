import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Album from "../components/album/Album";
import { AlbumContextProvider } from "../context/AlbumContext";
import { Switch } from "react-router-dom";

it("Landing component should render without crashing", () => {
  shallow(
    <Switch>
      <AlbumContextProvider>
        <Album />
      </AlbumContextProvider>
    </Switch>
  );
});

it("landing component snapshot with Enzyme", () => {
  const wrapper = shallow(
    <Switch>
      <AlbumContextProvider>
        <Album />
      </AlbumContextProvider>
    </Switch>
  );
  expect(toJson(wrapper)).toMatchSnapshot();
});
