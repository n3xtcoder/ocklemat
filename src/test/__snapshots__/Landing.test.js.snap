// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`landing component snapshot with Enzyme 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
